common/src/main/java/org/opensearch/ml/common/transport/mcpserver/requests/register/McpTool.java

```diff
@@ -0,0 +1,136 @@
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import lombok.Data;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ *  This class represents a tool that can be registered with OpenSearch. It contains information about the tool's name,
+ * description, parameters, and schema.
+ */
+@Log4j2
+@Data
+public class McpTool implements ToXContentObject, Writeable {
+    private static final String NAME_FIELD = "name";
+    private static final String DESCRIPTION_FIELD = "description";
+    private static final String PARAMS_FIELD = "params";
+    private static final String SCHEMA_FIELD = "schema";
+    private final String name;
+    private final String description;
+    private Map<String, Object> params;
+    private Map<String, Object> schema;
+    private static final String nameNotShownExceptionMessage = "name field required";
+
+    public McpTool(StreamInput streamInput) throws IOException {
+        name = streamInput.readString();
+        if (name == null) {
+            throw new IllegalArgumentException(nameNotShownExceptionMessage);
+        }
+        description = streamInput.readOptionalString();
+        if (streamInput.readBoolean()) {
+            params = streamInput.readMap(StreamInput::readString, StreamInput::readGenericValue);
+        }
+        if (streamInput.readBoolean()) {
+            schema = streamInput.readMap(StreamInput::readString, StreamInput::readGenericValue);
+        }
+    }
+
+    public McpTool(String name, String description, Map<String, Object> params, Map<String, Object> schema) {
+        if (name == null) {
+            throw new IllegalArgumentException(nameNotShownExceptionMessage);
+        }
+        this.name = name;
+        this.description = description;
+        this.params = params;
+        this.schema = schema;
+    }
+
+    public static McpTool parse(XContentParser parser) throws IOException {
+        String name = null;
+        String description = null;
+        Map<String, Object> params = null;
+        Map<String, Object> schema = null;
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            String fieldName = parser.currentName();
+            parser.nextToken();
+
+            switch (fieldName) {
+                case NAME_FIELD:
+                    name = parser.text();
+                    break;
+                case DESCRIPTION_FIELD:
+                    description = parser.text();
+                    break;
+                case PARAMS_FIELD:
+                    params = parser.map();
+                    break;
+                case SCHEMA_FIELD:
+                    schema = parser.map();
+                    break;
+                default:
+                    parser.skipChildren();
+                    break;
+            }
+        }
+        if (name == null) {
```

Comment:
As line 58 already checked null name, we can skip checking here

Reply:
Done